### PR TITLE
Spiderpalooza Webbing rework!

### PR DESCRIPTION
**Pull request by ghost**
:cl: Onule for sprites, Armhulen  coding
add: No longer do spiders stack infinite webs on one tile! There are wall and floor webs now, with the barricade webs restricted behind a caste for defenders!
tweak: Spiders no longer get stuck dragging victims on webs!
/:cl:

I needed to pr this early so dreamweaver can work with me on code, so don't expect much from what you see here. While spiders are more interesting than xenos to play against, they aren't quite as fun to play as. Hopefully this changes it making different variants more interesting twist and reworking bases to be cooler.

### CODE REVIEW

**Comment by Armhulen** [comment]
Er, i'm opening this for the thaw don't freak out!!

**Comment by Armhulen** [comment]
Onule is spriting for me thank god

**Comment by JJRcop** [comment]
For the future, you don't need to make a PR to collab with people, he can PR your branch as long as you upload it to github.

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 1]
```diff
@@ -155,7 +297,7 @@
 	speed = 1
 	gold_core_spawnable = NO_SPAWN
 
-//tarantulas are really tanky, regenerating (maybe), hulky monster but are also extremely slow, so.
+//tarantulas are really tanky, hulky monster but are also extremely off webs.
```
Comment (Anonmare) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: "Extremely off webs"? You mean "Extremely slow off webs"?

File: code/datums/status_effects/debuffs.dm [hunk 2]
```diff
@@ -81,6 +81,26 @@
...
+	var/newbrute = owner.getBruteLoss()
+	if(newbrute <= lastbrute)
+		var/brutediff = newbrute - lastbrute
+		owner.adjustBruteLoss(brutediff)
+	lastbrute = newbrute
+	var/newfire = owner.getFireLoss()
+	if(newfire <= lastfire)
+		var/firediff = newfire - lastfire
+		owner.adjustFireLoss(firediff)
+	lastfire = newfire
```
Comment (ninjanomnom) [inline code/datums/status_effects/debuffs.dm]: Really not sure how I feel about how this works. On the one hand mob code is a bitch and I understand it's a pain to do some things properly in there. On the other hand having a ticking operation to just undo changes elsewhere is disgusting.

Talk to me in coderbus sometime about doing this better. This needs to be changed to prevent healing rather than undoing previous changes.

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 3]
```diff
@@ -125,8 +125,156 @@
...
+/datum/action/innate/spider/ambush
+	name = "Tunnel"
+	desc = "Hide under the tiles, to ambush victims."
+	check_flags = AB_CHECK_CONSCIOUS
+	button_icon_state = "lay_web"
+
+/datum/action/innate/spider/ambush/Activate()
+	var/L = owner.loc
+	var/turf/T = get_turf(owner)
+	if(istype(owner.loc, /obj/effect/dummy/burrow))
```
Comment (ninjanomnom) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]:
```
var/obj/effect/dummy/burrow/B = owner.loc
var/turf/T = get_turf(owner)
if(istype(B))
  ...
```

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 4]
```diff
@@ -125,8 +125,156 @@
...
+		tunnelboy.setStaminaLoss(0, 0)
+		var/obj/effect/dummy/burrow/B2 = new(get_turf(tunnelboy.loc))
+		tunnelboy.forceMove(B2)
+		B2.jaunter = tunnelboy
+
+/obj/effect/dummy/burrow
+	name = "tremors"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "nothing"
+	var/canmove = 1
```
Comment (ninjanomnom) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: You don't seem to use this anywhere

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/datums/status_effects/debuffs.dm (hunk 2 above)
Comment (Armhulen) [reply code/datums/status_effects/debuffs.dm]: i'm trying out bitflags to see if that's better, one moment

**Comment by Dorsisdwarf** [comment]
add: Any sentient spiders may evolve to be the midwife, and pick special types for all spider types to evolve into!

uh

is this a joke? Any time xenobio sentient potions any spider/ a spider gets sentienced by the common sentience event, BOOM instant xeno-tier spider outbreak?

**Review by WJohn — APPROVED** [review]

**Comment by PKPenguin321** [comment]
Yeah this seems like a terrible idea

**Comment by Armhulen** [comment]
@Dorsisdwarf spiders are already on the level of xenos, except they can be countered by atmos. this change does not affect that. xenos can just make new queens where spiders get one ever to pick the castes.

i'm also lowering spider health to balance out coordination here

**Comment by Armhulen** [comment]
xdtm already make directives for new children, which is fine and actually makes adminning easier for us but it ruined the uncontrolled horde idea for spiders so i might as well try this

**Comment by PKPenguin321** [comment]
could you rephrase in a more coherent way

**Comment by Armhulen** [comment]
spiders r weaker then xenos (they die 2 atmos changes)

spiders still end da round fast

i nerf spiders too

**Comment by PKPenguin321** [comment]
ok but we were asking about xenobiology and how it sounds like this PR makes it so any xenobiologist can make a spiderpocalypse

**Comment by Armhulen** [comment]
giving one spider the option to remove all of it's offensive abilities to communicate with the team doesn't seem that bad. it can only send messages to the spiders, not receive. there can be infinite xeno queens. there can only be one midwife ever, if you kill it they can never get one again. compared to just making it rare, this is actually a nerf. i can decrease the amount of baby spiders hatched too if you'd like

**Comment by Armhulen** [comment]
spiders are simplemobs, they have no way to fix a lack of air and they will suicide people into escaping shocked doors and if they are in maint all you have to do is open up one wall from space and their entire base dies. if they're not in maint it's much easier to just siphon the one room.

**Comment by Armhulen** [comment]
and finally, how in the heck has xenobiology not been able to make a spider army for years now, i wasn't talking about xenobiology because this barely changes anything about it at all

**Comment by ExcessiveUseOfCobblestone** [comment]
Spawning eggs is the crux of their offensive (that can also spawn nurses which can also spawn....) so I'm not sure losing offensive capability is an entirely true statement.

Also I disagree with the whole "just open it to space" meme when they're sentient. They can reposition with the only thing that's nonrefundable (as of now) is the eggs themselves. That also has the nasty side effect of making the area unusuable, and that's assuming they don't bust open a door to then have air from actual hallways be slowly pulled out either.

Then again I don't expect this to be a common event so it'll probably be alright in actuality.

**Comment by Armhulen** [comment]
Considering what i'm doing to webs to make bases more important for spiders. I could also consider killing eggs if they have bad atmos.

**Comment by TheDreamweaver** [comment]
> I needed to pr this early so dreamweaver can work with me on code

Im so sorry @Armhulen

**Comment by Arianya** [comment]
I don't mind spiders as a xeno replacement but i feel like we've reached the far end of the slippery slope of giving the xenobiologist a personal army that's inextricably bound to his will as soon as he sentiences one queen/midwife

**Comment by Armhulen** [comment]
@Arianya respectable point but it was xdtm/someone else if i'm wrong's idea to let nurses and midwifes set an objective for their children and THAT'S what turned it into a personal army. What I liked about spiders is that they'd most certainly kill you too the moment they could.

**Comment by Armhulen** [comment]
IMO we should revert that but whatever

**Comment by ExcessiveUseOfCobblestone** [comment]
Please don't imply children of subverted parents should have had free reign to kill the parent's master.

Perhaps you should consider knocking off spiders from the pool and buffing up the event chances to be more common while also allowing for sentience if you want bad spiders but not bad spiders bound to a scientist.

**Comment by Armhulen** [comment]
ok, sure

i'm gonna stick to just the webbing changes in this pr though, that might make it less of a beetle battle to get in

**Comment by ExcessiveUseOfCobblestone** [comment]
You need to fix that copypasta of turf-checks.

**Comment by Armhulen** [comment]
explain

File: code/modules/reagents/chemistry/reagents/toxin_reagents.dm [hunk 5]
```diff
@@ -16,6 +16,27 @@
 		. = 1
 	..()
 
+/datum/reagent/toxin/antiheal
+	name = "Venomous Blight"
+	id = "spiderblight"
+	description = "A toxic venom that disrupts the body's ability to heal brute and burn injuries."
+	color = "#94ba57" // rgb: 148, 186, 87
+	taste_description = "bitterness"
```
Comment (iskyp) [inline code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: exact same taste description as /datum/reagent/toxin

File: code/modules/reagents/chemistry/reagents/toxin_reagents.dm [hunk 6]
```diff
@@ -16,6 +16,27 @@
...
+	taste_mult = 1.0
+	toxpwr = 1.5
```
Comment (iskyp) [inline code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: exact same toxpwr as /datum/reagent/toxin

File: code/modules/reagents/chemistry/reagents/toxin_reagents.dm [hunk 7]
```diff
@@ -16,6 +16,27 @@
 		. = 1
 	..()
 
+/datum/reagent/toxin/antiheal
+	name = "Venomous Blight"
+	id = "spiderblight"
+	description = "A toxic venom that disrupts the body's ability to heal brute and burn injuries."
+	color = "#94ba57" // rgb: 148, 186, 87
```
Comment (iskyp) [inline code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: uncapitalized letters, inconsistent with the rest of reagent code

**Review by ghost — CHANGES_REQUESTED** [review]
literally unplayable

**Comment by ExcessiveUseOfCobblestone** [comment]
https://github.com/Armhulen/tgstation/blob/bbef5d2f4e0bec547707d1284fc6b994974c8b76/code/modules/mob/living/simple_animal/hostile/giant_spider.dm#L181-L189

is almost identical to

https://github.com/Armhulen/tgstation/blob/bbef5d2f4e0bec547707d1284fc6b994974c8b76/code/modules/mob/living/simple_animal/hostile/giant_spider.dm#L181-L189

It can (should) be a single block of code with the minor differences captured via an argument.

**Comment by Armhulen** [comment]
moving to a new pr because i'm changing the goals of the pr and i already started work on it
